### PR TITLE
Begin fixing GoLint and Go Vet bugs

### PR DESCRIPTION
**Pull request by kkirsche**
Begin fixing bugs exposed by GoLint and Go Vet

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ language: go
 go:
   - 1.2
```
Comment (nemith) [inline .travis.yml]: 1.2 and 1.3 fail with the new ssh libs.  Lets remove them

File: netconf/jnpr/ioproc.go [hunk 2]
```diff
@@ -2,18 +2,21 @@ package jnpr
...
-	"github.com/Juniper/go-netconf/netconf"
 	"regexp"
+
+	"github.com/Juniper/go-netconf/netconf"
 )
 
 const (
-	NETCONF_CMD_CLI   = "junoscript netconf need-trailer"
-	NETCONF_CMD_SHELL = "exec xml-mode netconf need-trailer"
+	NetconfCMDCLI   = "junoscript netconf need-trailer"
```
Comment (nemith) [inline netconf/jnpr/ioproc.go]: These probably don't need to be exported. lowercase the first letter.

File: netconf/transport.go [hunk 3]
```diff
@@ -9,19 +9,24 @@ import (
 )
 
 const (
-	MSG_SEPERATOR = "]]>]]>"
+	// MsgSeperator is used to separate sent messages via NetConf
+	MsgSeperator = "]]>]]>"
```
Comment (nemith) [inline netconf/transport.go]: msgSeperator

File: netconf/transport.go [hunk 4]
```diff
@@ -84,8 +89,8 @@ func (t *transportBasicIO) WaitForFunc(f func([]byte) (int, error)) ([]byte, err
 	for {
 		n, err := t.Read(buf[pos : pos+(len(buf)/2)])
```
Comment (nemith) [inline netconf/transport.go]: We should probably check for the error up here which would solve this as well.

File: netconf/transport_ssh.go [hunk 5]
```diff
@@ -2,22 +2,28 @@ package netconf
...
 	"strings"
+
+	"golang.org/x/crypto/ssh"
 )
 
 const (
-	SSH_DEFAULT_PORT      = 830
-	SSH_NETCONF_SUBSYSTEM = "netconf"
+	// SSHDefaultPort is the default SSH port used when communicating with NetConf
+	SSHDefaultPort = 830
```
Comment (nemith) [inline netconf/transport_ssh.go]: sshDefaultPort

File: netconf/transport_ssh.go [hunk 6]
```diff
@@ -2,22 +2,28 @@ package netconf
...
+	// SSHNetconfSubsystem sets the SSH subsystem to NetConf
+	SSHNetconfSubsystem = "netconf"
```
Comment (nemith) [inline netconf/transport_ssh.go]: sshNetconfSubsystem

File: netconf/transport_ssh.go [hunk 7]
```diff
@@ -34,7 +40,7 @@ func (t *TransportSSH) Close() error {
 	return nil
 }
 
-// Dials and establishes an SSH sessions
+// Dial dials and establishes an SSH sessions
```
Comment (nemith) [inline netconf/transport_ssh.go]: Dial connects and establishes an SSH sessions

File: netconf/transport_telnet.go [hunk 8]
```diff
@@ -2,29 +2,37 @@ package netconf
...
 	"time"
+
+	"github.com/ziutek/telnet"
 )
 
 const (
-	TELNET_DEFAULT_PORT = 23
-	TELNET_TIMEOUT      = 10 * time.Second
+	// TelnetDefaultPort sets the default port for use by Telnet
+	TelnetDefaultPort = 23
```
Comment (nemith) [inline netconf/transport_telnet.go]: telnetDefaultPort

File: netconf/transport_telnet.go [hunk 9]
```diff
@@ -2,29 +2,37 @@ package netconf
...
+	// TelnetTimeout sets the timeout duration for use by Telnet
+	TelnetTimeout = 10 * time.Second
```
Comment (nemith) [inline netconf/transport_telnet.go]: telnetTimeout

**Comment by nemith** [comment]
Awesome work thanks for doing this.  I have some minor nits.

**Comment by kkirsche** [comment]
Fixed the majority of these and updated the README with the version requirements

**Pull request merged**